src/api/getSortedChangelogVersions.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+import { join as joinPath } from 'path';
+
+import { sync as syncGlob } from 'glob';
+
+import type {
+  ConfigType,
+  VersionType
+} from '../types';
+
+const VERSION_DIRECTORY_NAME_MATCH = /([0-9]+)\.([0-9]+)(\.([0-9]+))?$/;
+
+export default function getSortedChangelogVersions(config: ConfigType): VersionType[] {
+  return getVersionDirectoryNames(config)
+    .filter((versionDirectoryName) => !versionDirectoryName.endsWith('next'))
+    .map((versionDirectoryName) => {
+      const versionDirectoryMatch = versionDirectoryName.match(VERSION_DIRECTORY_NAME_MATCH);
+
+      if (!versionDirectoryMatch) {
+        throw new Error(
+          `Could not detect valid version on version directory ${versionDirectoryName}`
+        );
+      }
+
+      return versionDirectoryMatch;
+    }).map((versionMatch) => ({
+      major: parseInt(versionMatch[1], 10),
+      minor: parseInt(versionMatch[2], 10),
+      patch: versionMatch[4] ? parseInt(versionMatch[4], 10) : 0
```

Comment:
Wondering about what happened to versionMatch[3] :-) If you wanted to make it more obvious here, if you extracted some more variables